b2share/modules/schemas/views.py

```diff
@@ -139,12 +178,132 @@ def get(self, community_schema):
         return community_schema
 
 
+class BlockSchemaListResource(ContentNegotiatedMethodView):
+    view_name = 'block_schema_list'
+
+    def __init__(self, **kwargs):
+        """Constructor."""
+        super(BlockSchemaListResource, self).__init__(
+            method_serializers={
+                'GET': {
+                    'application/json': schemas_list_to_json_serializer,
+                },
+                'POST': {
+                    'application/json': new_schema_to_json_serializer,
+                }
+            },
+            default_method_media_type={
+                'GET': 'application/json',
+                'POST': 'application/json',
+            },
+            default_media_type='application/json',
+            **kwargs
+        )
+
+    def get(self):
+        """Get a list of all schemas."""
+        community_id = request.args['community_id']
+        schemas = BlockSchema.get_all_block_schemas(community_id=community_id)
+        return self.make_response(
+            schemas=schemas,
+            code=200)
+
+    def post(self):
+        """Create a new schema."""
+        if request.content_type != 'application/json':
+            abort(415)
```

Comment:
You can test it but I think `ContentNegotiatedMethodView` takes care of checking that the content-type is valid.